metrics/src/recorder/mod.rs

```diff
@@ -140,10 +140,16 @@ pub struct LocalRecorderGuard<'a> {
 impl<'a> LocalRecorderGuard<'a> {
     /// Creates a new `LocalRecorderGuard` and sets the thread-local recorder.
     fn new(recorder: &'a dyn Recorder) -> Self {
+        // SAFETY: Why this it okay to extend the lifetime of the object type from `'a` to `'static`?
```

Comment:
I think we can justify this as:

```suggestion
        // SAFETY: We extend `'a` to `'static` to satisfy the signature of `LOCAL_RECORDER`, which has an implied `'static` bound on
        // `dyn Recorder`, but we enforce that all usages of `LOCAL_RECORDER` are limited to `'a`, as we mediate their  access entirely through `LocalRecorderGuard<'a>`.
```